chef/ec-metal
# Death to the horrific private_chef_ha LWRP and bundler!

Sorry, this PR is massive and churny, but:
- Splits the private_chef_ha LWRP, which had grown huge and awful, back out into recipes plus a helper module
- Make Vagrant and EC2 machine_options config generation much more sane, pull it out of the vagrant/ec2 recipes
- Update to the latest chef-provisioning and friends
- remove Gemfile.lock and the bundler integration.  You should just use this with a recent ChefDK and stop complaining.
- Update to Chef 12 and the latest chef-provisioning-\* libraries
- reorganize and update the examples, so it's much clearer what is what
- tests!!!!
